var/spack/repos/builtin/packages/re2c/package.py

```diff
@@ -30,6 +30,10 @@ class Re2c(Package):
     phases = ["configure", "build", "install"]
 
     depends_on("cmake", when="platform=windows")
+    depends_on("gmake", when="platform=linux")
+    depends_on("gmake", when="platform=cray")
+    depends_on("gmake", when="platform=darwin")
+    depends_on("gmake", when="platform=freebsd")
```

Comment:
FWIW. You could use a for loop over the platforms and the context managers along the lines:

```suggestion
    for platform in ["linux", "cray", "darwin", "freebsd"]:
        with when(f"platform={platform}":
            depends_on("gmake")
```

Reply:
Doesn't save that much (one line) in this case.

Reply:
Yeah, I'm fine with either, but also like that `platform=freebsd` is `grep`'able